src/script/squirrel.hpp

```diff
@@ -50,6 +50,13 @@ class Squirrel {
 	/** Perform all the cleanups for the engine. */
 	void Uninitialize();
 
+	static SQUserPointer TypeTag(const char *tag)
+	{
+		static std::map<std::string, std::unique_ptr<uint8_t>> type_tags;
+		auto [it, _] = type_tags.insert({ tag, std::make_unique<uint8_t>() });
```

Comment:
Another option is:
```suggestion
		static std::map<std::string, std::unique_ptr<const char *>> type_tags;
		auto [it, _] = type_tags.insert({ tag, std::make_unique<const char *>(tag) });
```
Would allow to get and use the class name via `sq_gettypetag()` when needed. I'm thinking about implementing a mechanism to save/load some `OT_INSTANCE` elements.

Reply:
It must not be a `std::unique_ptr` as the object is already owned by something else. And there's no way to ensure that what it points to is still there.